pkg/envoy/lds/listener_test.go

```diff
@@ -166,3 +170,43 @@ func TestGetFilterMatchPredicateForTrafficMatches(t *testing.T) {
 		})
 	}
 }
+
+func TestNewOutboundListener(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	id := identity.K8sServiceAccount{}.ToServiceIdentity()
+	meshCatalog := catalog.NewMockMeshCataloger(mockCtrl)
+	meshCatalog.EXPECT().GetOutboundMeshTrafficPolicy(id).Return(&trafficpolicy.OutboundMeshTrafficPolicy{
+		TrafficMatches: []*trafficpolicy.TrafficMatch{
+			{
+				WeightedClusters: []service.WeightedCluster{{}},
+				DestinationIPRanges: []string{
+					"0.0.0.0/0",
+				},
+				DestinationPort:     1,
+				DestinationProtocol: constants.ProtocolTCPServerFirst,
+			},
```

Comment:
Could you also include a scenario in this test case where the protocol is not server first.

Reply:
I didn't test that here because this function calls `getFilterMatchPredicateForTrafficMatches()`, and the tests there cover that: https://github.com/openservicemesh/osm/blob/30fdefdf18b59b60084b9a485b5dc5696f47fc85/pkg/envoy/lds/listener_test.go#L131-L140